### PR TITLE
use LigerCrossEntropy loss with reduction=sum

### PR DESCRIPTION
**Pull request by thisisatharva-rh**
Currently, we are using `torch.nn.CrossEntropyLoss(reduction=sum)` with LK to facilitate multipack sampling, which leads to some weird behaviours as reported [here](https://issues.redhat.com/browse/RHELAI-4113). This PR enables us to use `LigerCrossEntropyLoss` with the multi-pack sampler, and solves the incorrect loss issue.

### CODE REVIEW

**Review by JamesKunstle — COMMENTED** [review]
This seems like a reasonable fix.

**Comment by thisisatharva-rh** [comment]
Closing due to some git-tree issues while rebasing, opened [#609](https://github.com/instructlab/training/pull/578) instead
